src/codeu/chat/client/commandline/Chat.java

```diff
@@ -89,7 +91,10 @@ private void doOneCommand(Scanner lineScanner) {
       if (!tokenScanner.hasNext()) {
         System.out.println("ERROR: No user name supplied.");
       } else {
-        signInUser(tokenScanner.nextLine().trim());
+        String userName = tokenScanner.nextLine().trim();
+        System.out.print("Please enter the password: ");
+        String password = lineScanner.next().trim();
```

Comment:
Sadly, I couldn't get it to work with the command line :( But, with the new private methods and switch cases, the class looks much better.